thread/source/lib.c

```diff
@@ -10,7 +10,7 @@ void *memset(void * ptr, int value, unsigned int num){
 }
 
 int strcomp(char* s1, char* s2, unsigned int size){
-	for(int i = 0; i < size; i++){
+	for(int i = 0; i < size*2; i++){	// doubling size brings this in line with strcopy() and fixes the "Rei" string randomly replacing "Ver" in other places.
```

Comment:
**WARN** Wouldn't this introduce some null-deref possibility? I mean it is where the checking should be enhanced, not the modification of such functions. I suggest you to make it test more chars after this strcmp is passed. Or use memcmp instead of this, so it would find 'Ver.\0' instead of just 'Ver.'. Hope this could be done in a safer way.